tests/react/vanilla-utils/freezeAtom.test.tsx

```diff
@@ -35,6 +35,64 @@ it('freezeAtom basic test', async () => {
   await screen.findByText('isFrozen: true')
 })
 
+it('freezeAtom handles null correctly', async () => {
+  const nullAtom = atom(null, (_get, set, _arg: null) => {
+    set(nullAtom, null)
+  })
+
+  const Component = () => {
+    const [value, setValue] = useAtom(freezeAtom(nullAtom))
+    return (
+      <>
+        <button onClick={() => setValue(null)}>set null</button>
+        <div>value is null: {`${value === null}`}</div>
+      </>
+    )
+  }
+
+  render(
+    <StrictMode>
+      <Component />
+    </StrictMode>,
+  )
+
+  await screen.findByText('value is null: true')
+
+  await userEvent.click(screen.getByText('set null'))
+  await screen.findByText('value is null: true')
+})
+
+it('freezeAtom handles primitive correctly', async () => {
+  const numberAtom = atom<number, [number], void>(123, (_get, set) => {
```

Comment:
```suggestion
  const numberAtom = atom(123, (_get, set, _arg: number) => {
```

@dai-shi How about this?

Reply:
this too.